docs/self-hosted-appcircle/install-server/helm-chart/configuration/docker-k8s-migration.md

```diff
@@ -0,0 +1,822 @@
+
+   ```bash
+   scp standalone-appcircle-server/app/appcircle-server/vaultd.tar.gz .
+   ```
+
+2. **Get the Vault statefulset name:**
+
+   ```bash
+   kubectl get statefulsets -n appcircle | grep vault
+   ```
+
+3. **Edit the vault `statefulset` for safe operations:**
+
+   ```bash
+   kubectl patch statefulset -n appcircle appcircle-server-vault -p '{"spec": {"template": {"spec":{"containers":[{"name":"vault","command": ["sh", "-c", "tail -f /dev/null" ], "args": null, "readinessProbe": null, "lifecycle": null  }]}}}}'
+   ```
+
+4. **Delete the pod for it to be re-created:**
+
+   ```bash
+   kubectl delete pod appcircle-server-vault-0 -n appcircle
+   ```
+
+5. **Copy the vault data to the target pod:**
+
+   ```bash
+   kubectl cp "./vaultd.tar.gz" "appcircle-server-vault-0:/vault/data/vaultd.tar.gz" -n appcircle
+   ```
+
+6. **Open shell in the vault container:**
+
+   ```bash
+   kubectl exec -it appcircle-server-vault-0 -- bash
+   ```
+
+7. **Run the following commands in the shell:**
+
+   ```bash
+   cd /vault/data
+   tar -xzvf vaultd.tar.gz -C .
+   /usr/local/bin/docker-entrypoint.sh vault server -config=/vault/config/extraconfig-from-values.hcl
+   ```
+
+8. **Don't close the upper terminal until the process finishes:**
+
+9. **Open a new terminal in the vault container:**
+
+   ```bash
+   kubectl exec -it appcircle-server-vault-0 -- bash
+   ```
+
+10. **Unseal the vault with the saved keys from the steps above:**
+
+    ```bash
+    vault operator unseal dnaDMnwLuRni******M0EPJ2gAlyeHmOAy
+    vault operator unseal FRTs/BO606ty******1nm9pJssLZjqVULR
+    vault operator unseal f35t4MU6gojw******/bH92wR9t6MzzIYc
+    ```
+
+11. **Delete the vault data tar ball:**
+
+    ```bash
+    rm /vault/data/vaultd.tar.gz
+    ```
+
+12. **Exit from the first and second vault terminal:**
+
+13. **Edit the secret with old unseal keys:**
+    ```bash
+    kubectl patch secret appcircle-server-vault-seal -n appcircle \
+    --patch='{"stringData": { "token": "*hvs*.U5LLy********F2bOy", "unseal_keys": "dnaDMnwLuRni******M0EPJ2gAlyeHmOAy FRTs/BO606ty******1nm9pJssLZjqVULR f35t4MU6gojw******/bH92wR9t6MzzIYc" }}'
+    ```
+
+## Post-Migration Steps
+
+After completing the migration, follow these steps to finalize and verify the setup:
+
+### 1. Update the `values.yaml`
+
+- **Remove Keycloak Replica Override:**  
+  Delete the `replicas: 0` setting under `auth-keycloak`.
+
+- **Remove MongoDB Resource Preset:**  
+  Delete the `resourcesPreset` setting under `mongodb` if you don't need it.
+
+### 2. Upgrade the Helm Release
+
+Apply the updated `values.yaml` configuration to the Appcircle server Helm chart:
+
+```bash
+helm upgrade --install appcircle-server appcircle/appcircle \
+  --timeout 1200s \
+  -n appcircle \
+  -f values.yaml
+```
+
+### 3. Update DNS Records
+
+- **Shift DNS to the New Cluster:**  
+  Update the DNS records to point to the new Appcircle Kubernetes cluster.
+
+- **Redis Subdomain Change:**  
```

Comment:
This step is optional. If user changes `redis` to `kvs` that he/she needs to update runner appsettings.json with the changed URL and restart runners.

If not, then no need runner maintenance.